File: bootcamp-2023/integration-testing/exercise/php/src/TripServiceKata/Trip/TripService.php
<?php

namespace TripServiceKata\Trip;

use Exception;
use TripServiceKata\User\User;
use TripServiceKata\User\UserSession;

class TripService
{
    private UserSession $session;

    public function __construct($userSession)
    {
        $this->session = $userSession;
    }

    public function getTripsByUser(User $user): array
    {
        $tripList = array();
        $loggedUser = $this->session->getLoggedUser();
        $isFriend = false;
        if ($loggedUser != null) {
            foreach ($user->getFriends() as $friend) {
                if ($friend == $loggedUser) {
                    $isFriend = true;
                    break;
                }
            }
            if ($isFriend) {
                $tripList = TripDAO::findTripsByUser($user);
            }
            return $tripList;
        } else {
            throw new Exception();
        }
    }
}

File: bootcamp-2023/integration-testing/exercise/c#-dotnet-5/TripServiceKata/TripServiceKata/Trip/TripService.cs
﻿using System.Collections.Generic;
using TripServiceKata.Exception;
using TripServiceKata.User;

namespace TripServiceKata.Trip
{
    public class TripService
    {
        private readonly UserSession session;
        public TripService(UserSession userSession)
        {
            session = userSession;
        }
        
        public List<Trip> GetTripsByUser(User.User user)
        {
            List<Trip> tripList = new List<Trip>();
            User.User loggedUser = session.GetLoggedUser();
            bool isFriend = false;
            if (loggedUser != null)
            {
                foreach(User.User friend in user.GetFriends())
                {
                    if (friend.Equals(loggedUser))
                    {
                        isFriend = true;
                        break;
                    }
                }
                if (isFriend)
                {
                    tripList = TripDAO.FindTripsByUser(user);
                }
                return tripList;
            }
            else
            {
                throw new UserNotLoggedInException();
            }
        }
    }
}
